src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskDatabaseServiceSearchTest.java

```diff
@@ -0,0 +1,356 @@
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_ordered_by_decs_task_list_and_count_when_search_find_some_tasks() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001",
+            "8d6cc5cf-c973-11eb-aaaa-100000000001",
+            "8d6cc5cf-c973-11eb-aaaa-200000000001",
+            "8d6cc5cf-c973-11eb-aaaa-300000000001",
+            "8d6cc5cf-c973-11eb-aaaa-400000000001",
+            "8d6cc5cf-c973-11eb-aaaa-000000000002",
+            "8d6cc5cf-c973-11eb-aaaa-000000000003",
+            "8d6cc5cf-c973-11eb-aaaa-000000000004");
+
+        SearchRequest searchRequest = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED, CFTTaskState.UNASSIGNED))
+            .jurisdictions(List.of("WA", "IA"))
+            .locations(List.of("765324", "765325"))
+            .roleCategories(List.of(RoleCategory.JUDICIAL, RoleCategory.CTSC))
+            .workTypes(List.of("hearing_work", "follow_up"))
+            .sortingParameters(List.of(new SortingParameter(SortField.CASE_NAME_CAMEL_CASE, SortOrder.DESCENDANT)))
+            .build();
+
+
+        GetTasksResponse<Task> response = cftTaskDatabaseService.searchForTasks(searchRequest,
+            accessControlResponse, false);
+        response.getTasks().forEach(t -> System.out.println(t.getId()));
+
+        assertEquals(8, response.getTotalRecords());
+        Assertions.assertThat(response.getTasks())
+            .hasSize(8)
+            .flatExtracting(Task::getId, Task::getCaseId)
+            .containsExactly(
+                newArrayList(
+                    "8d6cc5cf-c973-11eb-aaaa-400000000001", "1623278362440001",
+                    "8d6cc5cf-c973-11eb-aaaa-300000000001", "1623278362430001",
+                    "8d6cc5cf-c973-11eb-aaaa-200000000001", "1623278362420001",
+                    "8d6cc5cf-c973-11eb-aaaa-100000000001", "1623278362410001",
+                    "8d6cc5cf-c973-11eb-aaaa-000000000001", "1623278362400001",
+                    "8d6cc5cf-c973-11eb-aaaa-000000000002", "1623278362400002",
+                    "8d6cc5cf-c973-11eb-aaaa-000000000003", "1623278362400003",
+                    "8d6cc5cf-c973-11eb-aaaa-000000000004", "1623278362400004"
+                ).toArray()
+            );
+    }
+
+    @Test
+    void should_return_task_list_and_count_when_filter_task_by_state() {
+        List<RoleAssignment> roleAssignments = roleAssignmentsWithGrantTypeStandard();
+        AccessControlResponse accessControlResponse = new AccessControlResponse(userInfo, roleAssignments);
+        indexRecord("8d6cc5cf-c973-11eb-aaaa-000000000001", "8d6cc5cf-c973-11eb-aaaa-000000000002");
+
+        SearchRequest searchRequest = SearchRequest.builder()
```

Comment:
please review filter signature as some this test and some others use the exact same filter signature. is that the expectation?